### PR TITLE
Making origin field behavior consistent when modifying files via post-processing binaries

### PR DESCRIPTION
**Pull request by egthomas**
This pull request modifies several post-processing binaries to correctly set the `origin.code`, `origin.command`, and `origin.time` metadata fields.  The modified binaries are:

- lagfr_fix
- make_fit
- make_fitex
- make_fitex2
- make_lmfit
- make_raw
- make_sim
- sim_real
- trim_fit
- trim_iq
- trim_raw

Note that I also slightly modified the behavior of `trim_iq` to only print the extra record information when the `-vb` flag (rather than by default).

### CODE REVIEW

**Comment by mts299** [comment]
@egthomas can you please describe how these fields are changed, so I know how to test it and what to look for?

**Comment by egthomas** [comment]
The `origin` fields are in the radar parameter block of IQ, RAWACF, and FITACF files.  There are currently a lot of inconsistencies in how these fields are set when files are operated upon by binaries in the RST.  For example, `make_fit` correctly updates the `origin.command` and `origin.time` fields when a FITACF files is produced, but the `origin.code` field is not updated to indicate the file was not produced at a radar site.  This is a problem because sometimes FITACF (or other) files are produced on-site by radars and the `origin` fields are necessary to differentiate between them.

This pull request can be tested by comparing the `origin` fields of the `dmapdump` output from files operated on by the above routines.  On this branch the information will be correctly set by the most recent binary, while on the `develop` branch the three `origin` fields will not be handled consistently.

File: codebase/superdarn/src.bin/tk/tool/make_raw.1.8/make_raw.c [hunk 1]
```diff
@@ -214,11 +219,27 @@ int main (int argc,char *argv[]) {
     exit(-1);
   }
 
+  command[0]=0;
```
Comment (mts299) [inline codebase/superdarn/src.bin/tk/tool/make_raw.1.8/make_raw.c:222]: Can you explain in the comments what is going on here?

File: codebase/superdarn/src.bin/tk/tool/make_raw.1.8/make_raw.c [hunk 2]
```diff
@@ -214,11 +219,27 @@ int main (int argc,char *argv[]) {
...
+    if (c !=0) strcat(command," ");
+    strcat(command,argv[c]);
+  }
+
   while (IQFread(fp,prm,iq,&badtr,&samples) !=-1) {
     if (vb)
       fprintf(stderr,"%d-%d-%d %d:%d:%d beam=%d\n",prm->time.yr,prm->time.mo,
               prm->time.dy,prm->time.hr,prm->time.mt,prm->time.sc,prm->bmnum);
 
+    prm->origin.code=1;
```
Comment (mts299) [inline codebase/superdarn/src.bin/tk/tool/make_raw.1.8/make_raw.c:236]: Same here seems a lot more is going on in make_raw

**Review by mts299 — COMMENTED** [review]
just updating the code with some comments on checking if origin.code should be 1 or 0

**Comment by mts299** [comment]
Also can you merge `develop` into this branch, I cannot compile it otherwise:
make: *** [/home/marina/superdarn/rst_dev/rst//build/make/makelib.linux:25: stdkey] Error 1

**Comment by mts299** [comment]
looks good, After you fill in those 2 reviews I can merge it for you.

File: codebase/superdarn/src.bin/tk/tool/make_raw.1.8/make_raw.c (hunk 1 above)
Comment (ecbland) [reply codebase/superdarn/src.bin/tk/tool/make_raw.1.8/make_raw.c:222]:
```suggestion
 // Transfer command line arguments into "command" array (max 128 characters)
  command[0]=0;
```
@egthomas Is the `command[0]=0;` necessary? The code seems to work without it but I may have missed something here.

Comment (egthomas) [reply codebase/superdarn/src.bin/tk/tool/make_raw.1.8/make_raw.c:222]: In this pull request I've simply added the same code to handle the origin fields as is currently used in all of the other binaries (eg, [dattorawacf](https://github.com/SuperDARN/rst/blob/master/codebase/superdarn/src.bin/tk/reformat/dattorawacf.1.8/dattorawacf.c#L172), [fittofitacf](https://github.com/SuperDARN/rst/blob/master/codebase/superdarn/src.bin/tk/reformat/fittofitacf.1.8/fittofitacf.c#L156), [make_fit](https://github.com/SuperDARN/rst/blob/master/codebase/superdarn/src.bin/tk/tool/make_fit.1.19/make_fit.c#L314), [make_fitex](https://github.com/SuperDARN/rst/blob/master/codebase/superdarn/src.bin/tk/tool/make_fitex.1.4/make_fitex.c#L158), etc).  If someone wants to go back and update all of them to be consistent as far as comments or removal of certain lines then I don't see any problem.

File: codebase/superdarn/src.bin/tk/tool/make_raw.1.8/make_raw.c [hunk 3]
```diff
@@ -138,6 +138,11 @@ int main (int argc,char *argv[]) {
   int chnnum=0;
   int offset;
 
+  time_t ctime;
+  int c;
+  char command[128];
+  char tmstr[40];
```
Comment (mts299) [inline codebase/superdarn/src.bin/tk/tool/make_raw.1.8/make_raw.c:144]:
```suggestion
  // origin command to create the file to be stored in the file 
  char command[128];
  // time string for the ctime to be copied into as a string
  char tmstr[40];
```

File: codebase/superdarn/src.bin/tk/tool/make_raw.1.8/make_raw.c (hunk 2 above)
Comment (mts299) [inline codebase/superdarn/src.bin/tk/tool/make_raw.1.8/make_raw.c:236]:
```suggestion
    // set origin code to 1 as it is not being produced on site and copy it to file 
    prm->origin.code=1;
```

File: codebase/superdarn/src.bin/tk/tool/make_raw.1.8/make_raw.c [hunk 4]
```diff
@@ -214,11 +219,27 @@ int main (int argc,char *argv[]) {
     exit(-1);
   }
 
+  command[0]=0;
+  n=0;
+  for (c=0;c<argc;c++) {
+    n+=strlen(argv[c])+1;
+    if (n>127) break;
```
Comment (mts299) [inline codebase/superdarn/src.bin/tk/tool/make_raw.1.8/make_raw.c:226]:
```suggestion
    // if the commannd is too long just cut it off
    n+=strlen(argv[c])+1;
    if (n>127) break;
```
Should we make a warning? something to think about the future.

File: codebase/superdarn/src.bin/tk/tool/make_sim.1.0/make_sim.c [hunk 5]
```diff
@@ -75,7 +75,7 @@ void makeRadarParm(struct RadarParm * prm, char * argv[], int argc, int cpid, in
 
   prm->revision.major = 1;
   prm->revision.minor = 0;
-  prm->origin.code = 0;
+  prm->origin.code = 1;
```
Comment (mts299) [inline codebase/superdarn/src.bin/tk/tool/make_sim.1.0/make_sim.c:78]:
```suggestion
  // set to 1 as it is not produced on site 
  prm->origin.code = 1;
```

File: codebase/superdarn/src.bin/tk/tool/make_sim.1.0/make_sim_20120412.c [hunk 6]
```diff
@@ -75,7 +75,7 @@ void makeRadarParm(struct RadarParm * prm, char * argv[], int argc, int cpid, in
 
   prm->revision.major = 1;
   prm->revision.minor = 0;
-  prm->origin.code = 0;
+  prm->origin.code = 1;
```
Comment (mts299) [inline codebase/superdarn/src.bin/tk/tool/make_sim.1.0/make_sim_20120412.c:78]:
```suggestion
  // set to 1 as it is not produced on site 
  prm->origin.code = 1;
```

File: codebase/superdarn/src.bin/tk/tool/make_raw.1.8/make_raw.c [hunk 7]
```diff
@@ -214,11 +219,27 @@ int main (int argc,char *argv[]) {
...
+    if (c !=0) strcat(command," ");
```
Comment (mts299) [inline codebase/superdarn/src.bin/tk/tool/make_raw.1.8/make_raw.c:227]:
```suggestion
    // add spaces between commands then copy the command over to the origin command
    if (c !=0) strcat(command," ");
```

File: codebase/superdarn/src.bin/tk/tool/sim_real.1.0/sim_real.c [hunk 8]
```diff
@@ -83,7 +83,7 @@ void makeRadarParm2(struct RadarParm * prm, char * argv[], int argc, int cpid, i
 
   prm->revision.major = 1;
   prm->revision.minor = 0;
-  prm->origin.code = 0;
+  prm->origin.code = 1;
```
Comment (mts299) [inline codebase/superdarn/src.bin/tk/tool/sim_real.1.0/sim_real.c:86]:
```suggestion
  // set to 1 as it is not produced on site 
  prm->origin.code = 1;
```

File: codebase/superdarn/src.bin/tk/tool/make_raw.1.8/make_raw.c [hunk 9]
```diff
@@ -214,11 +219,27 @@ int main (int argc,char *argv[]) {
     exit(-1);
   }
 
+  command[0]=0;
+  n=0;
```
Comment (mts299) [inline codebase/superdarn/src.bin/tk/tool/make_raw.1.8/make_raw.c:223]: Is n defined somewhere in this file? Noticed below you did `int c,n;' but now here?

File: codebase/superdarn/src.bin/tk/tool/trim_fit.1.20/trim_fit.c [hunk 10]
```diff
@@ -156,6 +157,11 @@ int main (int argc,char *argv[]) {
   char *chnstr=NULL;
   char *cpstr=NULL;
 
+  time_t ctime;
+  int c,n;
+  char command[128];
+  char tmstr[40];
```
Comment (mts299) [inline codebase/superdarn/src.bin/tk/tool/trim_fit.1.20/trim_fit.c:163]:
```suggestion
  // origin time for when the file is produced
  time_t ctime;
  // counter for origin command length  
  int n=0;
  // origin command array to hold the string
  char command[128];
  // string to hold the origin time 
  char tmstr[40];
```

File: codebase/superdarn/src.bin/tk/tool/trim_fit.1.20/trim_fit.c [hunk 11]
```diff
@@ -292,36 +298,36 @@ int main (int argc,char *argv[]) {
...
- 
+
       if (set_channel==1) prm->channel=channel;
 
       if ((inxfp !=NULL) && (irec==1)) OldFitInxHeaderFwrite(inxfp,prm);
       dnum=OldFitFwrite(fp,prm,fit,NULL);
       if (inxfp !=NULL) OldFitInxFwrite(inxfp,drec,dnum,prm);
       atime=TimeYMDHMSToEpoch(prm->time.yr,
-	  	      prm->time.mo,
+                      prm->time.mo,
```
Comment (mts299) [inline codebase/superdarn/src.bin/tk/tool/trim_fit.1.20/trim_fit.c:308]: Any reason you change a bunch of spacing? maybe it should be aligned with `prm->time.yr,`?

File: codebase/superdarn/src.bin/tk/tool/trim_fit.1.20/trim_fit.c [hunk 12]
```diff
@@ -333,6 +339,15 @@ int main (int argc,char *argv[]) {
...
 
+    command[0]=0;
+    n=0;
+    for (c=0;c<argc;c++) {
+      n+=strlen(argv[c])+1;
+      if (n>127) break;
+      if (c !=0) strcat(command," ");
+      strcat(command,argv[c]);
+    }
+
```
Comment (mts299) [inline codebase/superdarn/src.bin/tk/tool/trim_fit.1.20/trim_fit.c:350]:
```suggestion
    // initialize array to be empty?
    command[0]=0;
    for (int c=0; c<argc; c++) {
      // check if the origin command is too long
      n+=strlen(argv[c])+1;
      // if so cut it off
      if (n>127) break;
      // add space between command line arguments and copy to origin command
      if (c !=0) strcat(command," ");
      strcat(command, argv[c]);
    }
```

File: codebase/superdarn/src.bin/tk/tool/trim_fit.1.20/trim_fit.c [hunk 13]
```diff
@@ -393,62 +408,48 @@ int main (int argc,char *argv[]) {
...
+                         yr,mo,dy,hr,mt,
+                         (int) sc,prm->channel,
+                         prm->bmnum,prm->cp);
+
+      prm->origin.code=1;
+      ctime= time((time_t) 0);
+      RadarParmSetOriginCommand(prm,command);
+      strcpy(tmstr,asctime(gmtime(&ctime)));
+      tmstr[24]=0;
+      RadarParmSetOriginTime(prm,tmstr);
```
Comment (mts299) [inline codebase/superdarn/src.bin/tk/tool/trim_fit.1.20/trim_fit.c:445]:
```suggestion
      // origin  code 1 means it is not produced on site
      prm->origin.code=1;
      // copy it over to the file
      ctime= time((time_t) 0);
      RadarParmSetOriginCommand(prm,command);
      strcpy(tmstr,asctime(gmtime(&ctime)));
      tmstr[24]=0;
      RadarParmSetOriginTime(prm,tmstr);
```

File: codebase/superdarn/src.bin/tk/tool/trim_iq.1.5/trim_iq.c [hunk 14]
```diff
@@ -126,7 +127,10 @@ int main (int argc,char *argv[]) {
 
   FILE *fp=NULL;
 
-  int n;
+  time_t ctime;
+  int c,n;
+  char command[128];
+  char tmstr[40];
```
Comment (mts299) [inline codebase/superdarn/src.bin/tk/tool/trim_iq.1.5/trim_iq.c:133]:
```suggestion
  // origin time for when the file is produced
  time_t ctime;
  // counter for origin command length  
  int n=0;
  // origin command array to hold the string
  char command[128];
  // string to hold the origin time 
  char tmstr[40];
```

File: codebase/superdarn/src.bin/tk/tool/trim_iq.1.5/trim_iq.c [hunk 15]
```diff
@@ -229,6 +233,15 @@ int main (int argc,char *argv[]) {
...
 
+  command[0]=0;
+  n=0;
+  for (c=0;c<argc;c++) {
+    n+=strlen(argv[c])+1;
+    if (n>127) break;
+    if (c !=0) strcat(command," ");
+    strcat(command,argv[c]);
+  }
+
```
Comment (mts299) [inline codebase/superdarn/src.bin/tk/tool/trim_iq.1.5/trim_iq.c:244]:
```suggestion
    // initialize array to be empty?
    command[0]=0;
    for (int c=0; c<argc; c++) {
      // check if the origin command is too long
      n+=strlen(argv[c])+1;
      // if so cut it off
      if (n>127) break;
      // add space between command line arguments and copy to origin command
      if (c !=0) strcat(command," ");
      strcat(command, argv[c]);
    }
```

File: codebase/superdarn/src.bin/tk/tool/trim_iq.1.5/trim_iq.c [hunk 16]
```diff
@@ -238,17 +251,26 @@ int main (int argc,char *argv[]) {
...
+      for (n=0;n<=prm->mplgs;n++) fprintf(stderr,"%d,%d ",prm->lag[0][n],prm->lag[1][n]);
+      fprintf(stderr,"\n");
+    }
+
+    prm->origin.code=1;
+    ctime = time((time_t) 0);
+    RadarParmSetOriginCommand(prm,command);
+    strcpy(tmstr,asctime(gmtime(&ctime)));
+    tmstr[24]=0;
+    RadarParmSetOriginTime(prm,tmstr);
```
Comment (mts299) [inline codebase/superdarn/src.bin/tk/tool/trim_iq.1.5/trim_iq.c:273]:
```suggestion
      // origin  code 1 means it is not produced on site
      prm->origin.code=1;
      // copy it over to the file
      ctime= time((time_t) 0);
      RadarParmSetOriginCommand(prm,command);
      strcpy(tmstr,asctime(gmtime(&ctime)));
      tmstr[24]=0;
      RadarParmSetOriginTime(prm,tmstr);
```

File: codebase/superdarn/src.bin/tk/tool/trim_raw.1.15/trim_raw.c [hunk 17]
```diff
@@ -137,6 +138,11 @@ int main (int argc,char *argv[]) {
 
   char vstring[256];
 
+  time_t ctime;
+  int c,n;
+  char command[128];
+  char tmstr[40];
+
```
Comment (mts299) [inline codebase/superdarn/src.bin/tk/tool/trim_raw.1.15/trim_raw.c:145]:
```suggestion
  // origin time for when the file is produced
  time_t ctime;
  // counter for origin command length  
  int n=0;
  // origin command array to hold the string
  char command[128];
  // string to hold the origin time 
  char tmstr[40];
```

File: codebase/superdarn/src.bin/tk/tool/trim_raw.1.15/trim_raw.c [hunk 18]
```diff
@@ -294,10 +300,19 @@ int main (int argc,char *argv[]) {
...
      }
+   } else {
+     command[0]=0;
+     n=0;
+     for (c=0;c<argc;c++) {
+       n+=strlen(argv[c])+1;
+       if (n>127) break;
+       if (c !=0) strcat(command," ");
+       strcat(command,argv[c]);
+     }
```
Comment (mts299) [inline codebase/superdarn/src.bin/tk/tool/trim_raw.1.15/trim_raw.c:315]:
```suggestion
    // initialize array to be empty?
    command[0]=0;
    for (int c=0; c<argc; c++) {
      // check if the origin command is too long
      n+=strlen(argv[c])+1;
      // if so cut it off
      if (n>127) break;
      // add space between command line arguments and copy to origin command
      if (c !=0) strcat(command," ");
      strcat(command, argv[c]);
    }
```

File: codebase/superdarn/src.bin/tk/tool/trim_raw.1.15/trim_raw.c [hunk 19]
```diff
@@ -306,18 +321,26 @@ int main (int argc,char *argv[]) {
...
        OldRawFwrite(fp,"rawwrite",prm,raw,recnum,NULL);
-     } else RawFwrite(stdout,prm,raw);
+     } else {
+       prm->origin.code=1;
+       ctime = time((time_t) 0);
+       RadarParmSetOriginCommand(prm,command);
+       strcpy(tmstr,asctime(gmtime(&ctime)));
+       tmstr[24]=0;
+       RadarParmSetOriginTime(prm,tmstr);
+       RawFwrite(stdout,prm,raw);
```
Comment (mts299) [inline codebase/superdarn/src.bin/tk/tool/trim_raw.1.15/trim_raw.c:342]:
```suggestion
      // origin  code 1 means it is not produced on site
      prm->origin.code=1;
      // copy it over to the file
      ctime= time((time_t) 0);
      RadarParmSetOriginCommand(prm,command);
      strcpy(tmstr,asctime(gmtime(&ctime)));
      tmstr[24]=0;
      RadarParmSetOriginTime(prm,tmstr);
```

**Review by mts299 — COMMENTED** [review]
@ecblandI did as @egthomas mentioned to add some comments since I am a firm believer in "leave a campsite cleaner than you found it" so this might help in the future for others who find the code else where.

File: codebase/superdarn/src.bin/tk/tool/make_raw.1.8/make_raw.c (hunk 1 above)
Comment (mts299) [reply codebase/superdarn/src.bin/tk/tool/make_raw.1.8/make_raw.c:222]: @egthomas if that is how you feel about commenting that is fine I did it for you. I figured this PR was about consistency to being well documented on that consistency would be good.

File: codebase/superdarn/src.bin/tk/tool/make_raw.1.8/make_raw.c (hunk 9 above)
Comment (ecbland) [reply codebase/superdarn/src.bin/tk/tool/make_raw.1.8/make_raw.c:226]: It's defined on line 114. I think it's ok to leave this code as-is
https://github.com/SuperDARN/rst/blob/967553525ea92ab2f64058ed519c5ddfb52f5ec5/codebase/superdarn/src.bin/tk/tool/make_raw.1.8/make_raw.c#L114

**Comment by ecbland** [comment]
OK, I think this is good to merge. Thanks @egthomas and @mts299 :100:

**Pull request merged**